src/Service/Resolvers/MsSqlQueryExecutor.cs

```diff
@@ -194,5 +197,59 @@ public override string GetSessionParamsQuery(HttpContext? httpContext, IDictiona
 
             return sessionMapQuery.ToString();
         }
+
+        /// <inheritdoc/>
+        public override async Task<DbResultSet> GetMultipleResultSetsIfAnyAsync(
+            DbDataReader dbDataReader, List<string>? args = null)
+        {
+            // From the first result set, we get the count(0/1) of records with given PK.
+            DbResultSet resultSetWithCountOfRowsWithGivenPk = await ExtractResultSetFromDbDataReader(dbDataReader);
+            int numOfRecordsWithGivenPK = (int)resultSetWithCountOfRowsWithGivenPk.Rows.FirstOrDefault()!.Columns[MsSqlQueryBuilder.COUNT_ROWS_WITH_GIVEN_PK]!;
```

Comment:
>(int)resultSetWithCountOfRowsWithGivenPk.Rows.FirstOrDefault()!.Columns[MsSqlQueryBuilder.COUNT_ROWS_WITH_GIVEN_PK]!; [](http://example.com/codeflow?start=42&length=117)

how do you guarantee that `FirstOrDefault()` and `Columns[MsSqlQueryBuilder.COUNT_ROWS_WITH_GIVEN_PK]` are not null?

Reply:
it will always be non-null because it is the count of rows with given PK. It will either be 0 or 1.

Reply:
What if the query fails to execute on the database?

Can we please use TryGet for the COUNT_ROWS_WITH_GIVEN_PK lookup as well?